### PR TITLE
Add caching of all BTF regardless of how it is sourced

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

- Add caching of BTF regardless of how it is sourced
- Flushes cached BTF upon demand, or 1 minute after last usage
- Adds fallback to embedded BTF search when platform is incorrectly detected (due to missing container mounts).
- Changes BTF load order to try default kernel BTF before embedded collection.

### Motivation

- Reduced work to obtain BTF
- Ensure memory used for BTF is not kept for process lifetime
- Find embedded BTF in more cases

### Additional Notes

The CO-RE/BTF loader is treated as a singleton for the process. If you want to change configuration, `pkg/ebpf.Reset()` needs to be called.

https://datadoghq.atlassian.net/browse/EBPF-244

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by guyarb — APPROVED** [review]
Looks good, if the change has any memory/cpu impact please add graphs from synthetic env or staging with the impact

**Review by hmahmood — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]
LGTM for files owned by @DataDog/container-integrations.

**Review by paulcacheux — APPROVED** [review]

**Review by usamasaqib — APPROVED** [review]

File: pkg/ebpf/btf.go [hunk 1]
```diff
@@ -100,62 +151,177 @@ func checkEmbeddedCollection(collectionPath string) (*btf.Spec, error) {
...
+	// try several ways to automatically find correct BTF file
+
+	// if we found a unique file within the collection, use that
+	if len(possiblePaths) == 1 {
+		pathParts := strings.Split(possiblePaths[0], string(os.PathSeparator))
+		if len(pathParts) > 0 {
+			if pathParts[0] != platform {
+				log.Warnf("BTF platform incorrectly detected as `%s`, using `%s` instead. Mount the /etc/os-release file as /host/etc/os-release in the system-probe container to resolve this warning", platform, pathParts[0])
+			}
+			if pathParts[0] == "ubuntu" && len(pathParts) > 2 && pathParts[1] != platformVersion {
```
Comment (val06) [inline pkg/ebpf/btf.go:228]: can you add a comment to explain why `ubuntu` use case has a special treatment?

**Review by val06 — APPROVED** [review]
added small comment

**Review by derekwbrown — APPROVED** [review]

**Pull request merged**